evm_arithmetization/src/generation/mod.rs

```diff
@@ -581,6 +585,9 @@ pub fn generate_traces<F: RichField + Extendable<D>, const D: usize>(
         mem_after: MemCap::default(),
     };
 
+    let use_keccak_tables =
+        !state.traces.keccak_inputs.is_empty() || !state.traces.keccak_sponge_ops.is_empty();
```

Comment:
nit: you can't have one empty and the other non-empty, so no need for the second evaluation